src/BluetoothManager.cpp

```diff
@@ -206,6 +206,7 @@ std::weak_ptr<BluetoothEvent> BluetoothManager::find(BluetoothType type,
 void BluetoothManager::handle_event(BluetoothType type, std::string *name,
     std::string *identifier, BluetoothObject *parent, BluetoothObject &object)
 {
+    std::lock_guard<std::mutex> guard(lock);
```

Comment:
Hi.

Is it possible, that add_event/remove_event API will be called in scope of callback (line 232). If so, crash will occur because 'event_list' will be modified during the iteration. To avoid this copy on write semantics should be implemented.